src/combinatorics/quiver/basic.lean

```diff
@@ -125,8 +125,74 @@ instance empty_quiver (V : Type u) : quiver.{u} (empty V) := ⟨λ a b, pempty
 
 @[simp] lemma empty_arrow {V : Type u} (a b : empty V) : (a ⟶ b) = pempty := rfl
 
-
 /-- A quiver is thin if it has no parallel arrows. -/
 @[reducible] def is_thin (V : Type u) [quiver V] := ∀ (a b : V), subsingleton (a ⟶ b)
 
+section push
+/-!
+### Pushing the quiver structure on `V` along a map `V → W`
+-/
+
+variables {V : Type*} [quiver V] {W : Type*} (σ : V → W)
+
+/-- The `quiver` instance obtained by pushing arrows of `V` along the map `σ : V → W` -/
+@[nolint unused_arguments]
+def push (σ : V → W) := W
+
+instance [h : nonempty W] : nonempty (push σ) := h
+
+/-- The quiver structure obtained by pushing arrows of `V` along the map `σ : V → W` -/
+@[nolint has_nonempty_instance]
+inductive push_quiver {V : Type u} [quiver.{v} V] {W : Type u₂} (σ : V → W) :
+  W → W → Type (max u u₂ v)
+| arrow {X Y : V} (f : X ⟶ Y) : push_quiver (σ X) (σ Y)
+
+instance : quiver (push σ) := ⟨push_quiver σ⟩
+
+namespace push
+
+/-- The prefunctor induced by pushing arrows via `σ` -/
+def of : V ⥤q push σ :=
+{ obj := σ,
+  map := λ X Y f, push_quiver.arrow f}
+
+@[simp] lemma of_obj : (of σ).obj = σ := rfl
+
+variables {W' : Type*} [quiver W'] (φ : prefunctor V W') (τ : W → W') (h : ∀ x, φ.obj x = τ (σ x) )
+
+include φ h
+/-- Any map `φ : V → W'` factoring through `τ : W → W'` lifts to a prefunctor from `V` to `W`. -/
```

Comment:
I named it `lift` mostly because I feel it corresponds to lifting of other kinds of maps (free structures, quotients, etc), and I think it's not misleading. As for `prefunctor.push`, that wouldn't be bad, but it's not obvious to me just from the name `prefunctor.push` what it refers to. I'd keep it as-is, but don't mind changing if it's clearly better in your opinion.